### PR TITLE
sdk unify with adapters

### PR DESCRIPTION
**Pull request by danstarns**
Related: https://github.com/rocket-connect/graphql-debugger/issues/185

Moves all data access to the client SDK and adapters.

### CODE REVIEW

File: packages/adapters/base/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# @graphql-debugger/adapters-base
```
Comment (danstarns) [inline packages/adapters/base/README.md:1]:
```suggestion
# @graphql-debugger/adapter-base
```

File: .github/workflows/quality.yml [hunk 2]
```diff
@@ -116,6 +116,22 @@ jobs:
...
+        run: cd ./packages/adapters/base && pnpm run test
+
+  test-adapter-sqlite:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-project
+      - name: graphql-debugger/adapter-base
+        run: cd ./packages/adapters/sqlite && pnpm run test
+
```
Comment (danstarns) [inline .github/workflows/quality.yml:134]: missing proxy

File: apps/collector-proxy/src/workers/post-schema.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export async function postSchemaWorker(data: PostSchema["body"]) {
 
     const hash = hashSchema(executableSchema);
 
-    const foundSchema = await prisma.schema.findFirst({
+    const foundSchema = await client.schema.findFirst({
```
Comment (akash-joshi) [inline apps/collector-proxy/src/workers/post-schema.ts:21]: @danstarns are we making the assumption here that the GQL Debugger client will have the same API as Prisma?

Comment (danstarns) [reply apps/collector-proxy/src/workers/post-schema.ts:21]: No, the api will mold itself as we add more connectors, but seeming that SQLite(prisma) is the only connector at this time its making it easier to port the code to the client.

File: package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0-alpha.77",
   "private": true,
   "scripts": {
-    "dev": "cross-env turbo dev --filter '!graphql-debugger' --filter '!@graphql-debugger/landing-page-app' --filter '!@graphql-debugger/landing-page-server' --concurrency=18",
+    "dev": "cross-env turbo dev --filter '!graphql-debugger' --filter '!@graphql-debugger/landing-page-app' --filter '!@graphql-debugger/landing-page-server' --concurrency=30",
```
Comment (akash-joshi) [inline package.json:6]: How does this concurrency bump help us?

File: packages/adapters/base/src/entities/Schema.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import {
```
Comment (akash-joshi) [inline packages/adapters/base/src/entities/Schema.ts:1]: I guess this explains my previous Q. Looks like we're following the Prisma API (as of 2024)

Comment (danstarns) [reply packages/adapters/base/src/entities/Schema.ts:1]: Pretty much at this time we are porting all the prisma calls with a direct sdk call using the conventions of `packages/schemas` & `pakages/graphql`.

Comment (danstarns) [reply packages/adapters/base/src/entities/Schema.ts:1]: As prisma is the only storage adapter at this point, however, the api will mold and change as we add more adapters. 

So this work is essantly just a big refactor to set us up for futuer adapt

File: packages/adapters/base/tests/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+describe("index", () => {
```
Comment (danstarns) [inline packages/adapters/base/tests/index.test.ts:1]: Base adapter needs testing

File: packages/adapters/proxy/tests/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+describe("index", () => {
```
Comment (danstarns) [inline packages/adapters/proxy/tests/index.test.ts:1]: Base proxy needs testing

File: packages/adapters/sqlite/tests/index.test.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+describe("index", () => {
```
Comment (danstarns) [inline packages/adapters/sqlite/tests/index.test.ts:1]: Base sqlite needs testing

File: packages/adapters/sqlite/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      span,
+      trace,
+    });
+
+    this.schema = schema;
+    this.span = span;
+    this.trace = trace;
+  }
+
+  public async clearDB(): Promise<boolean> {
```
Comment (danstarns) [inline packages/adapters/sqlite/src/index.ts:32]: Can we remove this?

File: .github/workflows/quality.yml [hunk 10]
```diff
@@ -116,6 +116,30 @@ jobs:
       - name: graphql-debugger/plugin-express
```
Comment (danstarns) [inline .github/workflows/quality.yml:116]: We should employ an matrix style testing where each adapter is tested.

**Review by danstarns — COMMENTED** [review]
Getting ready to merge

**Comment by danstarns** [comment]
Holding the merging of this in favor of [nextra docs](https://github.com/rocket-connect/graphql-debugger/pull/206)

This PR changes the surface area and introduces more components to debugger that need documenting.

**Pull request merged**